client/blocks/image-selector/README.md

```diff
@@ -0,0 +1,61 @@
+Image Selector
+===
+
+Image Selector is a component that uses the Media Modal to select images and preview previously selected images.  It allows multiple images to be passed or it can be limited to a single item. 
+
+## Usage
+
+Pass an array of image IDs that will be shown in the preview and preselected in the Media Modal.
+
+Changes to the selection of images will be passed back in various props for setting images, dropping images, and removing images.  See the example.jsx file for a more complete example.
+
+```jsx
+import ImageSelector from 'blocks/image-selector';
+
+export default function MyFeaturedImages() {
+	return (
+		<ImageSelector imageIds={ [ 34, 105 ] } multiple />
+	);
+}
+```
+
+This component needs both its own styles imported as well as the Media Modal styles on the page in which it's displayed.
+
+```scss
+@import 'assets/stylesheets/sections/media';
+@import 'blocks/image-selector/style';
+```
+
+### Props
+
+Props are displayed as a table with Name, Type, Default, and Description as headings.
+
+Name | Type | Default | Description
+--- | --- | --- | ---
+`className` | `string` | `` | String used for the class of outer ImageSelector wrapper
+`compact` | `bool` | `0` | Determines whether or not the upload picker is compact or full size
```

Comment:
Do you envision us using the `compact` prop anywhere?

Reply:
Definitely.  I've already used it in the branch that covers product variations as we always want this uploader compact since the product variation rows are pretty small.

Reply:
Sure thing.  Let me rebase and I'll push in a few minutes.

Reply:
@nb Here's the product image branch where you can see this use case in action https://github.com/Automattic/wp-calypso/pull/26703